demos/basic/actors/zmqActor.py

```diff
@@ -0,0 +1,172 @@
+from improv.actor import Actor
+
+import zmq
+from zmq import PUB, SUB, SUBSCRIBE, REQ, REP, LINGER, Again
+from zmq.log.handlers import PUBHandler
+import traceback
+
+import zmq.asyncio
+import asyncio
+import time
+
+import logging
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class ZmqPSActor(Actor):
+    """
+    Zmq actor with PUB/SUB pattern.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.send_socket = None
+        self.recv_socket = None
+        self.address = None
+        self.context = None
+
+    def setSendSocket(self, ip, port, timeout=0.001):
+        """
+        Sets up the send socket for the actor.
+        """
+
+        self.context = zmq.Context()
+        self.send_socket = self.context.socket(PUB)
+        # bind to the socket according to the ip and port
+        self.address = "tcp://{}:{}".format(ip, port)
+        self.send_socket.bind(self.address)
+        time.sleep(timeout)
+
+    def setRecvSocket(self, ip, port, timeout=0.001):
+        """
+        Sets up the receive socket for the actor.
+        """
+
+        self.context = zmq.Context()
+        self.recv_socket = self.context.socket(SUB)
+        self.address = "tcp://{}:{}".format(ip, port)
+        self.recv_socket.connect(self.address)
+        self.recv_socket.setsockopt(SUBSCRIBE, b"")
+        time.sleep(timeout)
+
+    def sendMsg(self, msg):
+        """
+        Sends a message to the controller.
+        """
+
+        self.send_socket.send_pyobj(msg)
```

Comment:
Probably worth a discussion here whether we want to default to sending Python Objects. It is less work on the user but may not be what people naively expect? I think I'm fine with it, though.